Sources/FrontEnd/TypeChecking/TypeChecker.swift

```diff
@@ -3627,87 +3625,75 @@ public struct TypeChecker {
     return r
   }
 
-  /// Returns the members of `decls` named `n`.
+  /// Returns `d` or an implementation thereof if it's named `n`. Otherwise, returns `nil`.
   ///
-  /// - Requires: The base name of all declarations in `decls` is equal to `n.stem`
-  mutating func filter(decls: DeclSet, named n: Name) -> [AnyDeclID] {
-    var matches = Array(decls)
-
-    // Filter out candidates whose argument labels do not match.
-    if !n.labels.isEmpty {
-      filter(decls: &matches, withLabels: n.labels)
+  /// - Requires: The base name of `d` is equal to `n.stem`
+  mutating func decl(_ d: AnyDeclID, named n: Name) -> AnyDeclID? {
+    if !n.labels.isEmpty && !decl(d, isLabeledWith: n.labels) {
+      return nil
     }
 
-    // Filter out candidates whose operator notation does not match.
-    if let notation = n.notation {
-      filter(decls: &matches, withNotation: notation)
+    if let notation = n.notation, !decl(d, hasNotation: notation) {
```

Comment:
```suggestion
    if let x = n.notation, operatorNotation(d) != x {
```
But I bet you don't even need to unwrap:

```suggestion
    if n.notation != operatorNotation(d) {
```

Reply:
Unfortunately I still need to unwrap because if `d` has a notation and `n` hasn't, then the method should return `d` anyway. That's why it was called "match" before: it's not a strict comparison.